From 0e5253950a96e35e546fe6a58ee81c685f904a21 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Wed, 16 Aug 2017 18:58:08 +0800
Subject: [PATCH 2387/5242] MLK-16197-12 video: fbdev: dcss: refine
 'dtg_channel_timing_config'

commit  db9205cf53d4e73f7472c4f28fb49ecb570288ed from
https://source.codeaurora.org/external/imx/linux-imx.git

Add 'blank' parameter to 'dtg_channel_timing_config' interface
to set or clear the channel display window according to the
blank state.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/imx_dcss.c |   29 ++++++++++++++++++++++-------
 1 file changed, 22 insertions(+), 7 deletions(-)

diff --git a/drivers/video/fbdev/mxc/imx_dcss.c b/drivers/video/fbdev/mxc/imx_dcss.c
index f8baa6c..9240805 100644
--- a/drivers/video/fbdev/mxc/imx_dcss.c
+++ b/drivers/video/fbdev/mxc/imx_dcss.c
@@ -1980,7 +1980,8 @@ static int dcss_dtg_start(struct dcss_info *info)
 	return 0;
 }
 
-static void dtg_channel_timing_config(struct dcss_channel_info *cinfo)
+static void dtg_channel_timing_config(int blank,
+				struct dcss_channel_info *cinfo)
 {
 	struct cbuffer *cb;
 	uint32_t ch_ulc_reg, ch_lrc_reg;
@@ -2011,10 +2012,24 @@ static void dtg_channel_timing_config(struct dcss_channel_info *cinfo)
 
 	cb = &cinfo->cb;
 
-	fill_sb(cb, chans->dtg_addr + ch_ulc_reg,
-		pos->ulc_y << 16 | pos->ulc_x);
-	fill_sb(cb, chans->dtg_addr + ch_lrc_reg,
-		pos->lrc_y << 16 | pos->lrc_x);
+	switch (blank) {
+	case FB_BLANK_UNBLANK:
+		/* set display window for one channel */
+		fill_sb(cb, chans->dtg_addr + ch_ulc_reg,
+			pos->ulc_y << 16 | pos->ulc_x);
+		fill_sb(cb, chans->dtg_addr + ch_lrc_reg,
+			pos->lrc_y << 16 | pos->lrc_x);
+		break;
+	case FB_BLANK_NORMAL:
+	case FB_BLANK_VSYNC_SUSPEND:
+	case FB_BLANK_HSYNC_SUSPEND:
+	case FB_BLANK_POWERDOWN:
+		fill_sb(cb, chans->dtg_addr + ch_ulc_reg, 0x0);
+		fill_sb(cb, chans->dtg_addr + ch_lrc_reg, 0x0);
+		break;
+	default:
+		return;
+	}
 }
 
 static void dtg_global_timing_config(struct dcss_info *info)
@@ -2068,7 +2083,7 @@ static int dcss_dtg_config(uint32_t ch_id, struct dcss_info *info)
 		dtg_global_timing_config(info);
 
 	/* TODO: Channel Timing Config */
-	dtg_channel_timing_config(cinfo);
+	dtg_channel_timing_config(FB_BLANK_UNBLANK, cinfo);
 
 	return 0;
 }
@@ -2767,7 +2782,7 @@ static int dcss_blank(int blank, struct fb_info *fbi)
 	struct dcss_info *info = cinfo->dev_data;
 	struct platform_device *pdev = info->pdev;
 
-	dtg_channel_timing_config(cinfo);
+	dtg_channel_timing_config(blank, cinfo);
 	dcss_channel_blank(blank, cinfo);
 
 #if USE_CTXLD
-- 
1.7.9.5

